pkg/controller/ksmtuned/node.go

```diff
@@ -0,0 +1,55 @@
+package ksmtuned
+
+import (
+	"fmt"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	ksmtunedv1 "github.com/harvester/node-manager/pkg/apis/node.harvesterhci.io/v1beta1"
+)
+
+func (c *Controller) NodeOnChange(_ string, node *corev1.Node) (*corev1.Node, error) {
+	if node == nil || node.DeletionTimestamp != nil {
+		return nil, nil
+	}
+
+	if node.Name != c.NodeName {
+		return node, nil
+	}
+
+	_, err := c.KsmtunedCache.Get(node.Name)
+	if apierrors.IsNotFound(err) {
```

Comment:
since err is nil in most time and we also need return node when err is not nil. prefer to
```go
	if _, err := c.KsmtunedCache.Get(node.Name); err != nil {
		if !apierrors.IsNotFound(err) {
			return node, err
		}
		if _, err = c.Ksmtuneds.Create(c.defaultKsmtuned(node)); err != nil {
			return node, fmt.Errorf("failed to create Ksmtuned: %s", err)
		}
	}
```